pkg/api/api.go

```diff
@@ -32,9 +33,9 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Get("/invite/:code", hs.Index)
 
 	// authed views
-	r.Get("/profile/", reqSignedIn, hs.Index)
-	r.Get("/profile/password", reqSignedIn, hs.Index)
-	r.Get("/profile/switch-org/:id", reqSignedIn, hs.ChangeActiveOrgAndRedirectToHome)
+	r.Get("/profile/", reqSignedInNonOAuth, hs.Index)
+	r.Get("/profile/password", reqSignedInNonOAuth, hs.Index)
+	r.Get("/profile/switch-org/:id", reqSignedInNonOAuth, hs.ChangeActiveOrgAndRedirectToHome)
```

Comment:
```suggestion
	r.Get("/profile/", reqSignedIn, hs.Index)
	r.Get("/profile/password", reqSignedInNonOAuth, hs.Index)
	r.Get("/profile/switch-org/:id", reqSignedIn, hs.ChangeActiveOrgAndRedirectToHome)
```

I don't understand why the user shouldn't be allowed to switch organization or change their own theme, time zone, and home dashboard just because they're not supposed to set a password for themselves.

Changing this would imply re-adding the profile option to the sidebar as well.